### PR TITLE
chore(connect): update coins.json support format

### LINKED ISSUE #8952: Add devices param to the supportedFirmware config in Connect
Currently there is no way to tie a Connect method to a specific device or firmware based on device. Extend the config to enable that functionality.

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

following changes in [trezor-common](https://github.com/trezor/trezor-common/commit/159a35070c9fef162197e8f50de4aabbf7bd91b4)

- `trezor1` renamed to `T1B1`
- `trezor2` renamed to `T2T1`
- added `T2B1`

using the same format in `AbstractMethod:FirmwareRange` interface:
- array [1: { min: 'version', max: 'version' }, 2: { min: 'version', max: 'version' }] becomes object 
```
{ 
  T1B1: { min: 'version', max: 'version' },
  T2T1: { min: 'version', max: 'version' },
  T2B1: { min: 'version', max: 'version' } 
}
```

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/8952

### CODE REVIEW

File: packages/connect/src/data/config.ts [hunk 1]
```diff
@@ -197,11 +197,11 @@ export const config = {
                 'setBusy',
                 'unlockPath',
             ],
-            min: ['1.12.1', '2.5.3'],
+            min: { T1B1: '1.12.1', T2T1: '2.5.3' },
         },
         {
             methods: ['showDeviceTutorial'],
-            min: ['0', '2.6.1'],
+            min: { T1B1: '0', T2T1: '0', T2B1: '2.6.1' },
```
Comment (mroz22) [inline packages/connect/src/data/config.ts:204]: ok, t2b1 only for specific cases, this config acts as a blacklist, so we dont need to specify t2b1 everywhere

**Comment by mroz22** [comment]
it looks ok 👍

**Comment by szymonlesisz** [comment]
as discussed IRL:
- i've moved custom FW range hardcoded in methods (rebootToBootloader, getFirmwareHash) to config
- i've verified and fixed `defaultFirmware` in `getAccountInfo` turns out the purpose of it is to tell (to [suite](https://github.com/trezor/trezor-suite/blob/develop/suite-common/wallet-core/src/discovery/discoveryThunks.ts#L516)) which index from the batch is unsupported. Example:
```
// requesting batch with T2B1 (model R)

TrezorConnect.getAccountInfo({
        bundle: [
            {
                coin: 'test',
                path: "m/84'/1'/0'",
            },
            {
                coin: 'vtc',
                path: "m/49'/28'/0'",
            },
        ],
});

// returns

{
  success: false,
  payload: {
    error: '[{"index":1,"exception":"ui-device_firmware_unsupported","coin":"vtc"}]',
    code: 'Method_Discovery_BundleException'
  }
}

// testnet on index 0 is supported but vtc on index 1 is not

```

File: packages/connect/src/data/config.ts [hunk 2]
```diff
@@ -197,12 +197,20 @@ export const config = {
...
             comment: ['Only on T2B1'],
         },
+        {
+            methods: ['rebootToBootloader'],
+            min: { T1B1: '1.11.0', T2T1: '2.6.0' },
+        },
+        {
+            methods: ['getFirmwareHash'],
+            min: { T1B1: '1.11.1', T2T1: '2.5.1' },
+        },
```
Comment (mroz22) [inline packages/connect/src/data/config.ts:214]: isn't authenticateDevice method missing in this list?

**Review by mroz22 — APPROVED** [review]
looks ok, just one comment (see above)

File: packages/connect/src/data/config.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/connect/src/data/config.ts:214]: it's added it different/previous commit:
https://github.com/trezor/trezor-suite/pull/9670/commits/6f40865ccea60fb593758a75f32c3e5eab088799

Comment (mroz22) [reply packages/connect/src/data/config.ts:214]: aaa, i missed that

**Pull request merged**